includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -228,6 +228,54 @@ private function filter_attributes( $attributes ) {
 		return $out;
 	}
 
+	/**
+	 * Parse the srcset attribute and deduplicate image candidates that have the same URL and width or pixel density
+	 * defined.
+	 *
+	 * @param DOMElement $node The img element.
+	 */
+	private function validate_srcset( $node ) {
```

Comment:
I think this should actually end up being put in `AMP_Tag_And_Attribute` sanitizer since it will currently only apply to `img` elements and not `amp-img` present in the original document, right?

Or maybe there should be a new `AMP_SrcSet_Sanitizer` similar to how there is a `AMP_Layout_Sanitizer`?

Aside: I just noticed that `AMP_Layout_Sanitizer` can currently be _unset_ via filters. It should be _mandatory_ to run just like `AMP_Meta_Sanitizer`, `AMP_Style_Sanitizer`, and `AMP_Tag_And_Attribute_Sanitizer` are.

As noted below, then the logic for sanitizing _and_ validating `srcset` could be contained in one sanitizer rather than being in both the `AMP_Img_Sanitizer` (which is really a _converter_) and the `AMP_Tag_And_Attribute_Sanitizer`.

Reply:
Made the `AMP_Layout_Sanitizer` mandatory in https://github.com/ampproject/amp-wp/pull/5366/commits/698f2454f499922ea2bfa8eb0fa5b97ffe9dc0ad.